src/partners/thorchain.ts

```diff
@@ -0,0 +1,198 @@
+import { datelog, smartIsoDateFromTimestamp } from '../util'
+
+const asThorchainTx = asObject({
+  date: asString,
+  in: asArray(
+    asObject({
+      address: asString,
+      coins: asArray(
+        asObject({
+          amount: asString,
+          asset: asString
+        })
+      ),
+      txID: asString
+    })
+  ),
+  out: asArray(
+    asObject({
+      address: asString,
+      coins: asArray(
+        asObject({
+          amount: asString,
+          asset: asString
+        })
+      ),
+      txID: asString
+    })
+  ),
+  pools: asArray(asString),
+  status: asString,
+  type: asString
+})
+
+const asApiKeys = asObject({
+  thorchainAddress: asString
+})
+
+const asSettings = asObject({
+  offset: asNumber
+})
+
+const asThorchainResult = asObject({
+  actions: asArray(asUnknown)
+})
+type Settings = ReturnType<typeof asSettings>
+type ThorchainResult = ReturnType<typeof asThorchainResult>
+
+const ROLLBACK = 500
+const LIMIT = 50
+const THORCHAIN_MULTIPLIER = 100000000
+
+export const queryThorchain = async (
+  pluginParams: PluginParams
+): Promise<PluginResult> => {
+  const ssFormatTxs: StandardTx[] = []
+
+  let offset = 0
+  let thorchainAddress
+  try {
+    thorchainAddress = asApiKeys(pluginParams.apiKeys).thorchainAddress
+    offset = asSettings(pluginParams.settings).offset
+  } catch (e) {}
+
+  if (thorchainAddress == null) {
+    return {
+      settings: { offset },
+      transactions: []
+    }
+  }
+
+  while (true) {
+    const url = `https://midgard.ninerealms.com/v2/actions?address=${thorchainAddress}&type=swap,refund&affiliate=ej&limit=50&offset=${offset}&limit=${LIMIT}`
+    let jsonObj: ThorchainResult
+    try {
+      const result = await fetch(url, {
+        method: 'GET'
+      })
+      const resultJson = await result.json()
+      jsonObj = asThorchainResult(resultJson)
+    } catch (e) {
+      datelog(e)
+      break
+    }
+    const txs = jsonObj.actions
+    for (const rawTx of txs) {
+      const {
+        date,
+        in: txIns,
+        out: txOuts,
+        pools,
+        status: txStatus
+      } = asThorchainTx(rawTx) // Check RAW trasaction
+
+      let status = 'complete'
+      if (txStatus !== 'success') {
+        status = 'failed'
+      }
+      if (pools.length !== 2) {
+        status = 'failed'
+      }
+
+      const srcAsset = txIns[0].coins[0].asset
+      const match = txOuts.find(o => {
+        const match2 = o.coins.find(c => c.asset === srcAsset)
+        return match2 != null
+      })
+
+      // If there is a match between source and dest asset that means a refund was made
+      // and the transaction failed
+      if (match != null) {
+        status = 'failed'
+      }
+
+      const timestampMs = div(date, '1000000', 16)
+      const { timestamp, isoDate } = smartIsoDateFromTimestamp(timestampMs)
+
+      const [chainAsset] = srcAsset.split('-')
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Why is this needed if `_` is used for the unused var? Perhaps try `_` alone instead of `_chain`?

Reply:
Neither works. Not sure what lint rule needs to change